lib/ansible/modules/cloud/amazon/ecr.py

```diff
@@ -0,0 +1,316 @@
+        name = params['name']
+        state = params['state']
+        policy = params['policy']
+        policy_text = params['policy_text']
+        delete_policy = params['delete_policy']
+
+        try:
+            if policy_text is not None:
+                # policy_text given; parse it
+                policy = json.loads(policy_text)
+            elif policy is not None:
+                # policy given; dump it
+                policy_text = json.dumps(policy)
+        except ValueError:
+            result['msg'] = 'Invalid JSON given for policy_text'
+            return False, result
+
+        result['state'] = state
+        result['created'] = False
+
+        repo = ecr.get_repository(name)
+
+        if state == 'present':
+            result['created'] = False
+            if not repo:
+                repo = ecr.create_repository(name)
+                result['changed'] = True
+                result['created'] = True
+            result['repository'] = repo
+
+            if delete_policy:
+                original_policy = ecr.get_repository_policy(name)
+
+                if verbosity >= 2:
+                    result['policy'] = None
+
+                if verbosity >= 3:
+                    result['original_policy'] = original_policy
+
+                if original_policy:
+                    ecr.delete_repository_policy(name)
+                    result['changed'] = True
+
+            elif policy is not None:
+                try:
+                    if verbosity >= 2:
+                        result['policy'] = policy
+                    original_policy = ecr.get_repository_policy(name)
+                    if verbosity >= 3:
+                        result['original_policy'] = original_policy
+
+                    if original_policy != policy:
+                        ecr.set_repository_policy(name, policy_text)
+                        result['changed'] = True
+                except:
+                    # Some failure w/ the policy. It's helpful to know what the
+                    # policy is.
+                    result['policy'] = policy_text
+                    raise
+
+        elif state == 'absent':
+            result['name'] = name
+            if repo:
+                ecr.delete_repository(name)
+                result['changed'] = True
+
+    except Exception as err:
+        msg = str(err)
+        if isinstance(err, ClientError):
+            msg = boto_exception(err)
+        result['msg'] = msg
+        result['exception'] = traceback.format_exc()
+        return False, result
+
+    if ecr.skipped:
+        result['skipped'] = True
+
+    if ecr.changed:
+        result['changed'] = True
+
+    return True, result
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True),
+        state=dict(required=False, choices=['present', 'absent'],
+                   default='present'),
+        policy=dict(required=False, type='dict'),
```

Comment:
This is a great use case for the `json` type that Ansible offers, so you can either pass valid JSON (such as from a file, or any other source) or a YAML dictionary defined in the task/play and Ansible will handle converting to a Python dict, eliminating the need for having both `policy` and `policy_text` options